app/Http/Controllers/HomeController.php

```diff
@@ -79,4 +80,29 @@ public function product()
         // dd($products);
         return view('index.product')->with('products', $products);
     }
+
+    /**
+    * Display product research
+    *
+    * @param Request $request [ value input tag ]
+    *
+    * @return [type]          [get all product event of search]
+    */
+    public function research(Request $request)
+    {
+        $category = $request->input('category');
+        $search = $request->input('search');
+        if (empty($category)) {
+              $category = "Product";
```

Comment:
[rank A] you are hardcode in this function so much. Please check and recover it again. Do you remember our policies? 
Product? Promotion? 16?